tests/unit/test_split_transfer.py

```diff
@@ -4,55 +4,300 @@
         mixed_transfers = [
             Transfer(
                 token=None,
-                receiver=barn_zerox,
+                receiver=self.solver,
                 amount_wei=amount_of_transfer,
             ),
-            Transfer(token=cow_token, receiver=barn_zerox, amount_wei=600 * ONE_ETH),
+            Transfer(
+                token=self.cow_token, receiver=self.solver, amount_wei=600 * ONE_ETH
+            ),
         ]
 
-        barn_slippage = SolverSlippage(
+        slippage = SolverSlippage(
             amount_wei=-amount_of_transfer - ONE_ETH,
-            solver_name="barn-0x",
-            solver_address=barn_zerox,
+            solver_name=self.solver_name,
+            solver_address=self.solver,
         )
-        indexed_slippage = {barn_zerox: barn_slippage}
-        accounting = SplitTransfersTest(period, mixed_transfers)
+        indexed_slippage = {self.solver: slippage}
+        accounting = SplitTransfers(self.period, mixed_transfers, PrintStore())
 
-        total_penalty = accounting.process_native_transfers(indexed_slippage)
+        total_penalty = accounting._process_native_transfers(indexed_slippage)
         expected_total_penalty = -amount_of_transfer
         self.assertEqual(total_penalty, expected_total_penalty)
 
+    def test_process_rewards(self):
+        cow_reward = 600 * ONE_ETH
+        mixed_transfers = [
+            Transfer(token=self.cow_token, receiver=self.solver, amount_wei=cow_reward),
+        ]
+        accounting = SplitTransfers(self.period, mixed_transfers, PrintStore())
+        reward_target = Address.from_int(7)
+        redirect_map = {
+            self.solver: Vouch(
+                solver=self.solver,
+                reward_target=reward_target,
+                bonding_pool=Address.zero(),
+            )
+        }
+        slippage_amount = 1
+        positive_slippage = [
+            SolverSlippage(
+                amount_wei=slippage_amount, solver_address=self.solver, solver_name=""
+            )
+        ]
+        accounting._process_rewards(redirect_map, positive_slippage)
+        # Although we haven't called process_native_transfers, we are appending positive slippage inside
+        self.assertEqual(
+            accounting.eth_transfers,
+            [
+                Transfer(
+                    token=None,
+                    receiver=redirect_map[self.solver].reward_target,
+                    amount_wei=slippage_amount,
+                )
+            ],
+        )
+        self.assertEqual(
+            accounting.cow_transfers,
+            [
+                Transfer(
+                    token=self.cow_token,
+                    receiver=redirect_map[self.solver].reward_target,
+                    amount_wei=cow_reward,
+                )
+            ],
+        )
+
+    def test_full_process(self):
+        """
+        This scenario involves two solvers, one with positive and one with negative slippage.
```

Comment:
Nit: From what I can tell these solvers don't share a reward target. Is it therefore necessary to test both sub-scenarios in one unit test (what aspect about having 2/3 solvers do we want to test compared to having 2/3 tests with one solver)? 
Generally, I find tests easier to understand if they just test a single thing with the minimum amount of code setup.

Reply:
That makes sense however, if we take this approach, then what is now two tests would become 6. I can proceed with that if you think its easier to follow. Tests would become:

single solver:
- positive slippage
- negative slippage not exceeding
- negative slippage exceeding ETH not COW
- negative slippage exceeding both

There is also room for other tests however (testing with & without redirects, testing multiple solvers with same reward target, etc...). 

Overall I like this suggestion, but I think that such a change might be better to include as a separate/follow-up PR so that it can be addressed more formally without cluttering this one. I will create an issue for this and follow up ASAP. Since this was only a nit from you I will assume you are OK with this.

Reply:
Yes, just a nit. It's imo a good [unit-test principle](https://tmap.net/node/699) . But feel free to use your own syle.

<img width="866" alt="image" src="https://user-images.githubusercontent.com/1200333/210999812-2c714f77-fa7d-4d2f-8718-17fdc1933ded.png">

Reply:
Done in https://github.com/cowprotocol/solver-rewards/pull/163/commits/5d6f6878aa6c161308827ee6469b4060bd43fe20 and was easy to change. Even looks good.

Reply:
Here is the follow up PR (based on this one) and can be merged into here directly if you'd like to see if all go through at once:

https://github.com/cowprotocol/solver-rewards/pull/171